semana4/lista-de-exercicios/script.js

```diff
@@ -0,0 +1,344 @@
+
+// Exercício 1
+// A função transforma o valor de uma moeda em dólar. 
+// Pede para o usuário digitar através do prompt e utiliza a função Number pra converter de sting para número.
+// Depois da conversão, printa no console o valor convertido em reais.
+
+// Exercício 2
+// A função investeDinheiro calcula o valorAposInvestimento conforme o parametro tipo de investimento do switch case. Depois disso vai ter o cálculo de dois montantes, só que o do "tesouro direto" vai dar erro, pois não há esse tipo de investimento.
+
+// Exercício 3
+// O for tem uma função que verifica se o número é par. Se for coloca no array1, caso contrário no 2. Depois ele ve o tamanho do array inicial "numeros" e depois dois dois novos arrays formados.
+
+//Exercício 4
+// Para cada número dentro do array números. O primeiro if não sei dizer, diria que vai dar um undefined por causa do infinity. Já o segundo, vai printar os números abaixo de zero, no caso o -10.
+
+// Lógica de programação
+
+//1) Fazendo um for com contador
+
+let contador;
+for(contador = 0; contador < 5; contador++){
+    console.log(contador);
+}
+
+// fazendo um while
+
+contador = 0
+while (contador > 5){
+    console.log(contador)
+    contador++
+}
+
+// for of
+
+let numeros = [1,2,4,6,8,9]
+for (let numero of numeros){
+    if (numero %2 === 0){
+        console.log("veio um par");
+    } else {
+        console.log ("veio um impar");
+    }
+}
+
+//2) a)false
+//   b) false
+//   c) true
+//   d)false
+//   e) true
+
+
+
+let quantidadeDeNumerosPares = 5
+let i = 0
+while(i < quantidadeDeNumerosPares) {
+  console.log((quantidadeDeNumerosPares*2)-2)
+  quantidadeDeNumerosPares--
+}
+
+// O código não funciona, tem uma execução sem fim. entao usei a quantidade de numeros pares decrescer, mas ainda imprimia o último, então coloquei o -2 como gambiarra...
+
```

Comment:
Você tem que escrever um código que, dado um número N, ele imprima (no console) os N primeiros números pares 
(por exemplo, se N for 3, você deve imprimir 0, 2 e 4).  Um colega seu disse que já começou esta tarefa, mas não conseguiu terminar. Dê uma olhada no código dele:
  const quantidadeDeNumerosPares
let i = 0
while(i <= quantidadeDeNumerosPares) {
  console.log(i*2)
}

Este código funciona? Por quê? Caso não funcione, corrija a implementação dele.RESPOSTA:

Não. Há dois erros principais:
  1) O Loop nunca vai para de rodar.Perceba que a condicação do loop é quando a variável i chegar no valor da variável quantidadeDeNumerosPares. Mesmo ela sendo declarada como "let", não há nenhuma linha de código que atualize o valor desta variável. Desta forma, o loop nunca vai parar de rodar.
2) A condição de parada está errada.Quando usamos while, o código dentro dele irá rodar até que a condição dentro dele se assuma o valor "false". Do jeito que está feito ele vai rodar uma vez a mais do que precisa.Vamos considerar que quantidadeDeNumerosPares seja 3. O loop ira imprimir no console:

// para i = 0
Console: 0

// para i = 1
Console: 2

// para i = 2
Console: 4

// para i = 3 (o código roda, porque i <= quantidadeDeNumerosPares é "true". i é igual a 3)
Console: 6

// para i = 4
// o loop vai parar de rodar porqu i <= quantidaddeDe NUmerosPares é "false". i é maior do que 3)

Uma possível correção do código está abaixo

```
function ex3NumerosPares() {
  const quantidadeDeNumerosPares
  let i = 0
  while (i < quantidadeDeNumerosPares) {
    console.log(i * 2)
    i++
  }
}
```